compose-lint-checks/src/main/java/slack/lint/compose/ViewModelInjectionDetector.kt

```diff
@@ -8,15 +8,28 @@ import com.android.tools.lint.detector.api.Issue
 import com.android.tools.lint.detector.api.JavaContext
 import com.android.tools.lint.detector.api.Severity
 import com.android.tools.lint.detector.api.SourceCodeScanner
+import com.android.tools.lint.detector.api.StringOption
 import org.jetbrains.kotlin.psi.KtCallExpression
 import org.jetbrains.kotlin.psi.KtFunction
 import org.jetbrains.kotlin.psi.KtProperty
 import slack.lint.compose.util.*
 import slack.lint.compose.util.sourceImplementation
 
-class ViewModelInjectionDetector : ComposableFunctionDetector(), SourceCodeScanner {
+class ViewModelInjectionDetector
+@JvmOverloads
+constructor(private val allowedNames: StringSetLintOption = StringSetLintOption(ALLOW_LIST)) :
+  ComposableFunctionDetector(allowedNames), SourceCodeScanner {
 
   companion object {
+
+    internal val ALLOW_LIST =
+      StringOption(
+        "allowed-viewmodel-injection",
```

Comment:
nit: since it's not really an allow-list, let's improve the semantics. Maybe `USER_FACTORIES` and `viewmodel-factories`?

Reply:
[d39522a](https://github.com/slackhq/compose-lints/pull/53/commits/d39522accc202e1ecd2dc5c89558607b1560f716)